test/integration/patients/worklist/worklist.js

```diff
@@ -2635,6 +2635,108 @@ context('worklist page', function() {
       .should('contain', 'Patient Field B');
   });
 
+  specify('flow sorting numerical - patient field', function() {
+    cy
+      .routeSettings(fx => {
+        const sortingSettings = _.find(fx.data, setting => setting.id === 'sorting');
+        _.each(sortingSettings.attributes.value, function(sortMethod) {
+          sortMethod.sort_type = 'numeric';
+        });
+
+        return fx;
+      })
+      .routeFlows(fx => {
+        fx.data = _.sample(fx.data, 3);
+
+        fx.data[0].relationships.patient = { data: { id: '2' } };
+        fx.data[1].relationships.patient = { data: { id: '3' } };
+        fx.data[2].relationships.patient = { data: { id: '1' } };
+
+        fx.included.push({
+          id: '1',
+          type: 'patients',
+          attributes: {
+            first_name: 'Patient',
+            last_name: 'Field 1',
+          },
+          relationships: { 'patient-fields': { data: [{ id: '1' }] } },
+        });
+
+        fx.included.push({
+          id: '1',
+          type: 'patient-fields',
+          attributes: { value: { value: 1 }, name: 'foo' },
+        });
+
+        fx.included.push({
+          id: '2',
+          type: 'patients',
+          attributes: {
+            first_name: 'Patient',
+            last_name: 'Field None',
+          },
+        });
+
+        fx.included.push({
+          id: '3',
+          type: 'patients',
+          attributes: {
+            first_name: 'Patient',
+            last_name: 'Field 2',
+          },
+          relationships: { 'patient-fields': { data: [{ id: '3' }] } },
+        });
+
+        fx.included.push({
+          id: '3',
+          type: 'patient-fields',
+          attributes: { value: { value: 2 }, name: 'foo' },
+        });
+
+        return fx;
+      })
+      .visit('/worklist/shared-by')
+      .wait('@routeFlows');
+
+    cy
+      .get('.worklist-list__filter-sort')
+      .click()
+      .get('.picklist')
+      .contains('Foo: Highest - Lowest')
+      .click();
+
+    cy
+      .get('.app-frame__content')
+      .find('.table-list__item')
+      .first()
+      .should('contain', 'Patient Field 2');
+
+    cy
+      .get('.app-frame__content')
+      .find('.table-list__item')
+      .last()
+      .should('contain', 'Patient Field None');
+
+    cy
+      .get('.worklist-list__filter-sort')
+      .click()
+      .get('.picklist')
+      .contains('Foo: Lowest - Highest')
+      .click();
+
+    cy
+      .get('.app-frame__content')
+      .find('.table-list__item')
+      .first()
+      .should('contain', 'Patient Field None');
+
+    cy
+      .get('.app-frame__content')
+      .find('.table-list__item')
+      .last()
+      .should('contain', 'Patient Field 2');
+  });
```

Comment:
To create this test, I copied over the `"flow sorting alphabetical - patient field"` test and changed the custom sorting org setting to have `sort_type: 'numeric'` for both sorting methods (`asc` and `desc`).

Then I updated the patient field data to have numercial values instead of alphabetical (e.g. `1` vs. `A`). And changed the patient names to have numbers in them to make the test easier to understand (`Patient Field 1` vs. `Patient Field A`).